lib/media/source_buffer_manager.js

```diff
@@ -439,6 +471,46 @@ shaka.media.SourceBufferManager.prototype.clear_ = function() {
 
 
 /**
+ * Clear the source buffer after the given timestamp (aligned to the next segment boundary).
+ *
+ * @param {number} timestamp
+ *
+ * @return {!Promise}
+ * @private
+ */
+shaka.media.SourceBufferManager.prototype.clearAfter_ = function(timestamp) {
+    shaka.asserts.assert(!this.operationPromise_);
+
+    if (this.sourceBuffer_.buffered.length == 0) {
+        shaka.log.v1('Nothing to clear.');
+        shaka.asserts.assert(this.inserted_.length == 0);
+        return Promise.resolve();
+    }
+
+    var segmentIndex = shaka.media.SegmentReference.find(this.inserted_, timestamp)
```

Comment:
Suggest renaming |segmentIndex| to |index| or just |i| as we use |segmentIndex| elsewhere to refer to ISegmentIndex instances.